### PR TITLE
sqlccl: correctly mutate table descriptors during resumed RESTORE

### PR DESCRIPTION
**Pull request by madelynnblue**
If a RESTORE was resumed after a pause or node failure, it would
attempt to regenerate the sql descriptors it was using. However it did
this in a way that was different than the first invocation that didn't
use the same resume codepath. This mutation was to remove the FK on
indexes if the user requested it. Move the mutation to a downstream
place so both invocation paths will mutate index descriptors in the
same way.

Also improve some test infra to return errors more quickly and with
useful messages.

### CODE REVIEW

**Comment by mjibson** [comment]
Ping @dt.

File: pkg/ccl/sqlccl/backup_test.go [hunk 1]
```diff
@@ -619,18 +619,32 @@ func TestBackupRestoreCheckpointing(t *testing.T) {
...
+		}
 		if e, a := jobs.StatusSucceeded, jobs.Status(status); e != a {
 			return errors.Errorf("expected backup status %s, but got %s", e, a)
 		}
 		return nil
 	})
+	if jobFailedErr != nil {
+		return jobFailedErr
+	}
+	return err
```
Comment (benesch) [inline pkg/ccl/sqlccl/backup_test.go]: Ah, much better!

**Review by benesch — APPROVED** [review]
Nice, LGTM!

**Comment by dt** [comment]
:lgtm: nice catch!

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20092#-:-KzPqpZ9DjMvYjVgZzcK:b-pe67a)*

**Pull request merged**